packages/nest/src/generators/init/lib/add-dependencies.ts

```diff
@@ -1,27 +1,56 @@
 import type { GeneratorCallback, Tree } from '@nrwl/devkit';
-import { addDependenciesToPackageJson } from '@nrwl/devkit';
+import { addDependenciesToPackageJson, readJson } from '@nrwl/devkit';
+import { satisfies } from 'semver';
 import {
-  nestJsSchematicsVersion,
-  nestJsVersion,
+  nestJsSchematicsVersion7,
+  nestJsSchematicsVersion8,
+  nestJsVersion7,
+  nestJsVersion8,
   nxVersion,
   reflectMetadataVersion,
-  rxjsVersion,
+  rxjsVersion6,
+  rxjsVersion7,
 } from '../../../utils/versions';
 
 export function addDependencies(tree: Tree): GeneratorCallback {
+  // Old nest 7 and rxjs 6 by default
+  let NEST_VERSION = nestJsVersion7;
+  let NEST_SCHEMATICS = nestJsSchematicsVersion7;
+  let RXJS = rxjsVersion6;
+
+  const packageJson = readJson(tree, 'package.json');
+
+  if (packageJson.dependencies['@angular/common']) {
```

Comment:
suggestion/nitpick: let's use `angular/core` instead of `angular/common`. Though it doesn't really matter in this case, but we feel like `angular/core` is a tad better. That's why it's a "nitpick" ^^